### PR TITLE
cleaned and fixed mash.rule

### PR DESCRIPTION
**Pull request by RNAer**
also cleaned up the master Snakefile

### CODE REVIEW

File: oecophylla/distance/mash.rule [hunk 1]
```diff
@@ -12,137 +12,26 @@ rule mash_sketch:
...
     threads:
         1
     log:
         distance_dir + "logs/mash_sketch.sample=[{sample}].log"
     benchmark:
-        "benchmarks/distance/mash_sketch.sample=[{sample}].json"
-    run:
-        output_base = os.path.splitext(output['sketch'])[0]
+        "benchmarks/distance/mash_sketch.sample=[{sample}].txt"
+    shell:
```
Comment (wasade) [inline oecophylla/distance/mash.rule]: previously this was `run`?

File: oecophylla/distance/mash.rule [hunk 2]
```diff
@@ -12,137 +12,26 @@ rule mash_sketch:
...
+        # -r is for reads instead of genomes for mash sketch;
+        # remove suffix because mash will automatically add .msh
+        '''set +u; {params.env}; set -u
+        output_basename=`basename {output[0]} .msh`
```
Comment (wasade) [inline oecophylla/distance/mash.rule:22]: why `output[0]`, is that always the first sample?

File: oecophylla/distance/mash.rule [hunk 3]
```diff
@@ -12,137 +12,26 @@ rule mash_sketch:
...
         distance_dir + "logs/mash_sketch.sample=[{sample}].log"
     benchmark:
-        "benchmarks/distance/mash_sketch.sample=[{sample}].json"
-    run:
-        output_base = os.path.splitext(output['sketch'])[0]
+        "benchmarks/distance/mash_sketch.sample=[{sample}].txt"
+    shell:
+        # -r is for reads instead of genomes for mash sketch;
+        # remove suffix because mash will automatically add .msh
+        '''set +u; {params.env}; set -u
```
Comment (wasade) [inline oecophylla/distance/mash.rule:21]: does `shell` interpret a newline as a character return?

**Review by wasade — COMMENTED** [review]
Just some questions

File: oecophylla/distance/mash.rule (hunk 3 above)
Comment (RNAer) [reply oecophylla/distance/mash.rule:21]: yes if it is triple quoted

File: oecophylla/distance/mash.rule (hunk 1 above)
Comment (RNAer) [reply oecophylla/distance/mash.rule]: yes. it was python and now bash code only

File: oecophylla/distance/mash.rule (hunk 2 above)
Comment (RNAer) [reply oecophylla/distance/mash.rule:22]: yes. actually only using forward reads.

Comment (RNAer) [reply oecophylla/distance/mash.rule:22]: note it's for output - there is only one output file needed and specified.

**Comment by mortonjt** [comment]
This needs to be pushed against the dev branch

**Pull request closed without merging**